core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -807,21 +809,20 @@ VariableValue make_variable_value(size_t id, double val) {
   variable_value.value.double_value = val;
   return variable_value;
 }
-Experimental::VariableValue make_variable_value(size_t id, const char* val) {
+Experimental::VariableValue make_variable_value(size_t id,
+                                                const std::string& val) {
   VariableValue variable_value;
-  variable_value.type_id            = id;
-  variable_value.value.string_value = val;
+  variable_value.type_id = id;
+  strncpy(variable_value.value.string_value, val.c_str(), 63);
   return variable_value;
 }
-SetOrRange make_candidate_set(size_t size, const char** data) {
+SetOrRange make_candidate_set(size_t size, std::string* data) {
   SetOrRange value_set;
-  const char** data_copy = new const char*[size];
+  value_set.set.values.string_value = new tuning_string[size];
```

Comment:
I held back from commenting above for `variable_value.value.string_value` but `value_set.set.values.string_value` are you for real?

Reply:
Also would you remind me who is in charge for calling delete?